manifests/virtualenv.pp

```diff
@@ -99,10 +99,16 @@
     } elsif (( versioncmp($_virtualenv_version,'1.7') < 0 ) and ( $systempkgs == false )) {
       $system_pkgs_flag = '--no-site-packages'
     } else {
-      $system_pkgs_flag = $systempkgs ? {
-        true    => '--system-site-packages',
-        false   => '--no-site-packages',
-        default => fail('Invalid value for systempkgs. Boolean value is expected')
+      case $systempkgs {
+        true: { $system_pkgs_flag = '--system-site-packages' }
+        false: {
+          if ( versioncmp($_virtualenv_version, '20.0.1') < 0 ) {
```

Comment:
Why not add another `elsif` a few lines up?

```pp
    } elsif (( versioncmp($_virtualenv_version,'20.0.1') > 0 ) and ( $systempkgs == false )) {
      $system_pkgs_flag = ''
```

Tbh that logic structure isn't too clean either, but at least it's consistently dirty.

Reply:
i think i agree with @thebigb on this one
this `case` is pretty unclean in the else, so why not avoid it altogether

Reply:
Ah there's actually a subtle difference in logic here. When the `virtualenv_version` is not yet populated (e.g. the package has been just installed in the current run), @saz's version will default to an empty `$system_pkgs_flag`, and my version will default to `--no-site-packages`.

I don't know enough about older versions virtualenv. If between 1.7 and 20.0.1 the `--no-site-packages` option was redundant anyway, the logic proposed by this PR is favorable. Otherwise this breaks the first run for everyone installing versions < 20.0.1.

Reply:
So depending on what's desirable one of these should do the trick:

```pp
    if (( versioncmp($_virtualenv_version,'1.7') > 0 ) and ( $systempkgs == true )) {
      $system_pkgs_flag = '--system-site-packages'
    } elsif (( versioncmp($_virtualenv_version,'20.0.1') < 0 ) and ( $systempkgs == false )) {
      $system_pkgs_flag = '--no-site-packages'
    } else {
      $system_pkgs_flag = $systempkgs ? {
        true    => '--system-site-packages',
        false   => '',
        default => fail('Invalid value for systempkgs. Boolean value is expected')
      }
    }
```

```pp
    if (( versioncmp($_virtualenv_version,'1.7') > 0 ) and ( $systempkgs == true )) {
      $system_pkgs_flag = '--system-site-packages'
    } elsif (( versioncmp($_virtualenv_version,'1.7') < 0 ) and ( $systempkgs == false )) {
      $system_pkgs_flag = '--no-site-packages'
    } elsif (( versioncmp($_virtualenv_version,'20.0.1') > 0 ) and ( $systempkgs == false )) {
      $system_pkgs_flag = ''
    } else {
      $system_pkgs_flag = $systempkgs ? {
        true    => '--system-site-packages',
        false   => '--no-site-packages',
        default => fail('Invalid value for systempkgs. Boolean value is expected')
      }
    }
```

Reply:
Found a bug in the examples of my previous comment. For some reason I thought that it would jump to the `else` block if the `virtualenv_version` is empty, but that's obviously not the case. I edited the comment and the logic should work now.

Reply:
> I don't know enough about older versions virtualenv. If between 1.7 and 20.0.1 the `--no-site-packages` option was redundant anyway, the logic proposed by this PR is favorable. Otherwise this breaks the first run for everyone installing versions < 20.0.1.
Looking at https://virtualenv.pypa.io/en/legacy/changes.html#v1-7-2011-11-30 it's the case starting from 1.7